ClangSharpPInvokeGenerator/CXTypeExtensions.cs

```diff
@@ -0,0 +1,408 @@
+﻿using System.Diagnostics;
+using ClangSharp;
+
+namespace ClangSharpPInvokeGenerator
+{
+    internal static class CXTypeExtensions
```

Comment:
If the writer is extended to support user-configuration, these might need to either take additional parameters or become instance methods on the writer... The later might be better since they are more specific to the writer than to the actual interop types.